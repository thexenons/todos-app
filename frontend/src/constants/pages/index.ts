import { lazy } from "react";

import { PageKey, Pages } from "../../types";

export const PAGES: Pages = {
	[PageKey.home]: {
		path: "/",
		title: "Home",
		component: lazy(() => import("../../pages/Home")),
	},
	[PageKey.login]: {
		path: "/login",
		component: lazy(() => import("../../pages/Login")),
	},
	[PageKey.register]: {
		path: "/register",
		component: lazy(() => import("../../pages/Register")),
	},

	[PageKey.test]: {
		path: "/test",
		title: "Test",
		loader: async () => {
			return await (await import("../../pages/Test")).getInitialData();
		},
		component: lazy(() => import("../../pages/Test")),
		isProtected: true,
		children: {
			[PageKey.todosListSingle]: {
				path: "/:id",
				title: "Todos List Single",
				loader: async (args) => {
					return await (
						await import("../../pages/Test/TodosListSingle")
					).getInitialData(args);
				},
				component: lazy(() => import("../../pages/Test/TodosListSingle")),
			},
		},
	},
	[PageKey.components]: {
		path: "/components",
		title: "Components",
		component: lazy(() => import("../../pages/Components")),
		hideLayout: true,
		children: {
			[PageKey.componentsButton]: {
				path: "/button",
				title: "Button",
				component: lazy(() => import("../../pages/Components/Button")),
			},
			[PageKey.componentsAcceptButton]: {
				path: "/accept-button",
				title: "AcceptButton",
				component: lazy(() => import("../../pages/Components/AcceptButton")),
			},
			[PageKey.componentsUpdateButton]: {
				path: "/update-button",
				title: "UpdateButton",
				component: lazy(() => import("../../pages/Components/UpdateButton")),
			},
			[PageKey.componentsCreateButton]: {
				path: "/create-button",
				title: "CreateButton",
				component: lazy(() => import("../../pages/Components/CreateButton")),
			},
			[PageKey.componentsDeleteButton]: {
				path: "/delete-button",
				title: "DeleteButton",
				component: lazy(() => import("../../pages/Components/DeleteButton")),
			},
			[PageKey.componentsCancelButton]: {
				path: "/cancel-button",
				title: "CancelButton",
				component: lazy(() => import("../../pages/Components/CancelButton")),
			},
			[PageKey.componentsPanel]: {
				path: "/panel",
				title: "Panel",
				component: lazy(() => import("../../pages/Components/Panel")),
			},
			[PageKey.componentsModal]: {
				path: "/modal",
				title: "Modal",
				component: lazy(() => import("../../pages/Components/Modal")),
			},
			[PageKey.componentsFloatingPanel]: {
				path: "/floating-panel",
				title: "FloatingPanel",
				component: lazy(() => import("../../pages/Components/FloatingPanel")),
			},
			[PageKey.componentsInputBase]: {
				path: "/input-base",
				title: "InputBase",
				component: lazy(() => import("../../pages/Components/InputBase")),
			},
		},
	},
};
