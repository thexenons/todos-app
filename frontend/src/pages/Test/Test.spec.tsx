import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { vi } from "vitest";

import BaseTestComponent from "../../tests/BaseTestComponent";
import { TEST_TODOS_LIST } from "../../tests/constants";
import Test from ".";

vi.mock("react-router-dom", async () => {
	const reactRouterDom = await vi.importActual<Record<string, unknown>>(
		"react-router-dom"
	);

	return {
		...reactRouterDom,
		useLoaderData: () => ({
			todosLists: [TEST_TODOS_LIST],
		}),
	};
});

describe("Test", () => {
	afterAll(() => {
		vi.restoreAllMocks();
		fetchMock.mockClear();
	});

	it("Render", () => {
		const { container } = render(
			<BaseTestComponent>
				<Test />
			</BaseTestComponent>
		);

		expect(container).toMatchSnapshot();
	});

	it("onTodosListClick", async () => {
		render(
			<BaseTestComponent>
				<Test />
			</BaseTestComponent>
		);

		const clickableLists = screen.getAllByRole("presentation");

		expect(clickableLists).toHaveLength(1);

		for (const clickableList of clickableLists) {
			await userEvent.click(clickableList);
		}
	});
});
