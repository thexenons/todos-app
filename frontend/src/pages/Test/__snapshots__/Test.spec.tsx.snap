// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test > Render 1`] = `
<div>
  <div
    class="styled__ContainerWrapper-sc-1kleavz-0 iIEyCq"
  >
    <h1>
      Test
    </h1>
    <div>
      <div>
        <h2>
          Todos List 
        </h2>
        <ul>
          <li
            role="presentation"
          >
            <h3>
              test todos list
            </h3>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
